### PR TITLE
Fix scrapeconfigs syntax

### PR DESCRIPTION
**Pull request by bd3lage**
**How to categorize this PR?**

/area monitoring
/kind regression

**What this PR does / why we need it**:
Fixes https://github.com/gardener/gardener/pull/9688#pullrequestreview-2090892830

The regression was introduced with `v1.95.0`

**Special notes for your reviewer**:

I was not able to test this with the local-setup as it seems it does not use gardener-operator to install the observability stack. Let me know if you can assist with testing.
cc @istvanballok 

**Release note**:

```bugfix operator
The broken `prometheus-longterm` `ScrapeConfig`s introduced with `v1.95.0` have been fixed.
```

### CODE REVIEW

File: pkg/component/observability/monitoring/prometheus/longterm/scrapeconfigs.go [hunk 1]
```diff
@@ -56,8 +56,8 @@ func CentralScrapeConfigs() []*monitoringv1alpha1.ScrapeConfig {
 						`{__name__=~"garden_shoot_info:timestamp:this_month"}`,
 						`{__name__=~"metering:(cpu_requests|memory_requests|network|persistent_volume_claims|disk_usage_seconds|memory_usage_seconds).*:this_month"}`,
 						`{__name__="garden_shoot_node_info"}`,
-						`{__name__="garden_shoot_condition", condition=("APIServerAvailable"|"SystemComponentsHealthy")}`,
-						`{__name__="garden_seed_condition", condition=("SeedSystemComponentsHealthy"|"GardenletReady")}`,
+						`{__name__="garden_shoot_condition", condition="(APIServerAvailable|SystemComponentsHealthy)"}`,
+						`{__name__="garden_seed_condition", condition="(SeedSystemComponentsHealthy|GardenletReady)"}`,
```
Comment (istvanballok) [inline pkg/component/observability/monitoring/prometheus/longterm/scrapeconfigs.go]:
```suggestion
						`{__name__="garden_shoot_condition", condition=~"(APIServerAvailable|SystemComponentsHealthy)"}`,
						`{__name__="garden_seed_condition", condition=~"(SeedSystemComponentsHealthy|GardenletReady)"}`,
```

**Review by istvanballok — CHANGES_REQUESTED** [review]
Thank you @bd3lage for taking care to fix the regression that was introduced by

> federate additional metrics in the Prometheus longterm instance #9688

Please use the `kind` `regression` in the PR description and mention that it was introduced with [v1.95.0](https://github.com/gardener/gardener/releases/tag/v1.95.0) so that the colleagues can request the cherry-pick PRs from Prow.

Please also add a release note. Release notes are typically not needed if a regression is fixed before a version was released that contains the bug. For this specific case though, `v1.95` and `v1.96` are affected, so we should have a release note.

Note that unfortunately we need to take extra care when we change monitoring related configuration and we need to make sure that the PR achieves its intention. I mean with regular golang coding, the go compiler helps to prevent syntax issues and the various lint checks and unit tests make sure to have a high quality source code. Monitoring configuration on the other hand is just a string in the source code, so we need to take care ourselves to prevent syntax errors (like in #9688) and to achieve the intention of the PR (like in this fix).

Before opening a PR, I suggest to try out the change in the local setup or tweak the dev landscape configuration to gain confidence that the PR is working. Otherwise, if this effort is not feasible, opening an issue instead of a PR is maybe better. If a PR is opened, at least I assume that the author is confident that the change is working.

**Review by istvanballok — COMMENTED** [review]
The diff looks good to me.
Before I could give a `lgtm`, please don't forget to adjust the PR description as I described above, and please also describe how you tested your change.
It is fine to explicitly say that the change was not tested in the local setup if you did not have the opportunity to do so.

**Review by rfranzke — COMMENTED** [review]
/approve

**Review by istvanballok — APPROVED** [review]
> I was not able to test this with the local-setup as it seems it does not use gardener-operator to install the observability stack.

Please check this document: [docs/concepts/operator.md#local-development](https://github.com/gardener/gardener/blob/master/docs/concepts/operator.md#local-development)

> Let me know if you can assist with testing.

I tested the change. Thank you for providing the fix!

/lgtm

Note that in the gardener-operator local setup I referenced above, you can check the syntax error in the federation configuration of the long term prometheus.

With #9763, we can do even more! 🎉 
We can create a shoot and verify that after the introduction of the new federation instant vector selectors of #9688:

```text
    - '{__name__="garden_shoot_condition", condition=~"(APIServerAvailable|SystemComponentsHealthy)"}'
    - '{__name__="garden_seed_condition", condition=~"(SeedSystemComponentsHealthy|GardenletReady)"}'
    - '{__name__="garden_seed_usage"}'
    - '{__name__="garden_seed_capacity"}'
    - '{__name__="etcdbr_snapshot_duration_seconds_count"}'
```

the long term prometheus federates some more metrics from the garden prometheus, that are originally exposed by the gardener metrics exporter.

Even if this PR is going to be merged now, I recommend testing this in the local setup, just for practice (and for future PRs). Thank you.

**Comment by istvanballok** [comment]
I'm wondering why Prow did not open a cherry-pick PR for the `v1.95` branch, although there was a comment above `cherry-pick release-v1.95` which Prow even acknowledged.

https://github.com/gardener/gardener/pulls?q=is%3Apr+%22Fix+scrapeconfigs+syntax%22

I double checked that the fix is relevant for `v1.95`.

**Pull request merged**